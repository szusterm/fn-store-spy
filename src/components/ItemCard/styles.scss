@import '../../scss/variables';
@import '../../scss/shadow';
@import '../../scss/transition';

.item-card {
	$strip-height: 16px;

	cursor: pointer;

	max-width: 100%;
	height: auto;
	margin: 10px;

	&--top-part, &--bot-part {
		background-color: $body-bg;
		@include shadow(sm);

		@include transition(sh);
		transition-property: box-shadow;
	}

	&:hover {
		.item-card--top-part, .item-card--bot-part {
			@include shadow(md);
		}
	}

	&--top-part {
		padding: 12px;

		border-top-left-radius: 50%;
		border-top-right-radius: 50%;

		position: relative;

		&::after {
			width: 100%;
			background: $body-bg;
			height: $strip-height;
			content: "";

			display: block;
			position: absolute;
			top: calc(100% - #{$strip-height}/2);
			left: 0;
			right: 0;
		}
	}

	&--bot-part {
		$padding: 15px;

		height: 110px;
		padding-top: $strip-height/2;

		border-bottom-left-radius: 10px;
		border-bottom-right-radius: 10px;

		display: flex;
		flex-direction: column;
		justify-content: space-between;
		align-items: center;

		.bot-part {
			color: $body-color;
			text-align: center;
			font-family: 'Roboto', Arial;

			&--name {
				font-size: 1.1em;
				font-weight: 500;
				text-align: center;
				padding: 0 $padding;
			}

			&--description {
				width: 100%;
				padding: $padding;

				display: flex;
				flex-direction: row;
				justify-content: space-around;
				align-items: center;

				.description {
					&--type {
						font-size: 0.8em;
						font-weight: 400;
					}

					&--price {
						font-size: 1em;
						font-weight: 100;
					}
				}
			}
		}
	}
}