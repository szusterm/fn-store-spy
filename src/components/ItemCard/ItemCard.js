import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {addItemToOrder, setActiveInOrderedItem} from '../../redux/actions';

import './styles.scss';

export class ItemCard extends Component {
	constructor() {
		super();

		this.switchOrder = this.switchOrder.bind(this);
	}

	switchOrder() {
		const {_id, name, rarity} = this.props;

		if (this.checkThatItemIsOrdered()) {
			const item = {
				_id,
				name,
				rarity
			};

			this.props.addItemToOrder(item);
		}
		else {
			const indexInOrder = this.getIndexInOrder();
			//method to remove item from order
		}
	}

	checkThatItemIsOrdered() {
		const {_id} = this.props;
		const {orderedItems} = this.props;

		const sameItemInOrder = orderedItems.find((orderedItem) => (_id === orderedItem._id));

		return (sameItemInOrder !== undefined);
	}

	getIndexInOrder() {
		const {_id} = this.props;
		const {orderedItems} = this.props;

		let indexInOrder = null;

		orderedItems.find((orderedItem, orderedItemIndex) => {
			if (_id === orderedItem._id) {
				indexInOrder = orderedItemIndex;
			}
		});

		return indexInOrder;
	}

	render() {
		const {name, price, rarity, type, imageSrc} = this.props;

		return (
			<div
				className={'item-card item-card--raised'}
				onClick={this.switchOrder}
			>
				<div className={'item-card--top-part'}>
					<div className={`top-part--img-box top-part--img-box--${rarity}`}>
						<img
							className={'img-box--image'}
							src={imageSrc}
						/>
					</div>
				</div>
				<div className={'item-card--bot-part'}>
					<div className={'bot-part--name'}>{name}</div>
					{
						(this.checkThatItemIsOrdered()) &&
							<span>ORDERED</span>
					}
					<div className={'bot-part--description'}>
						<div className={'description--type'}>{type}</div>
						<div className={'description--price'}>{price}</div>
					</div>
				</div>
			</div>
		);
	}
}

ItemCard.propTypes = {
	_id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	price: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	imageSrc: PropTypes.string.isRequired,
	orderedItems: PropTypes.array, //redux,
	addItemToOrder: PropTypes.func, //redux
	setActiveInOrderedItem: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {items: orderedItems} = state.order;
	return {orderedItems};
};

const mapDispatchToProps = {addItemToOrder, setActiveInOrderedItem};

export default connect(mapStateToProps, mapDispatchToProps)(ItemCard);