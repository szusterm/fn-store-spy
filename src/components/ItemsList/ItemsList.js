import React, {Component} from 'react';
import PropTypes from 'prop-types';

import ItemCard from '../ItemCard';

import './styles.scss';

class ItemsList extends Component {
	render() {
		const {items} = this.props;

		return (
			<div>
				{
					items.map((item) => {
						const {key, name, price, rarity, imgSrc} = item;

						return (
							<ItemCard
								key={key}
								name={name}
								price={price}
								rarity={rarity}
								imgSrc={imgSrc}
							/>
						);
					})
				}
			</div>
		);
	}
}

ItemsList.propTypes = {
	items: PropTypes.array
};

export default ItemsList;