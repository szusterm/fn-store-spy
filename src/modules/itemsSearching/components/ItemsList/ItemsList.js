import React, {Component} from 'react';
import PropTypes from 'prop-types';

import ItemCard from '../ItemSearchCard';

import './styles.scss';

export class ItemsList extends Component {
	render() {
		const {items} = this.props;

		return (
			<div className={'col-12 col-sm-12 col-md-11 col-lg-10 col-xl-8'}>
				<div className={'container-fluid'}>
					<div className={'row no-gutter'}>
						{
							items.map((item) => {
								const {_id, fnbrId, name, price, rarity, type, imageSrc} = item;

								return (
									<ItemCard
										key={_id}
										fnbrId={fnbrId}
										name={name}
										price={price}
										rarity={rarity}
										type={type}
										imageSrc={imageSrc}
									/>
								);
							})
						}
					</div>
				</div>
			</div>
		);
	}
}

ItemsList.propTypes = {
	items: PropTypes.array
};

export default ItemsList;