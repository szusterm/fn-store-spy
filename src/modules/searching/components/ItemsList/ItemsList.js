import React, {Component} from 'react';
import PropTypes from 'prop-types';

import ItemCard from '../ItemSearchCard/index';

import './styles.scss';

export class ItemsList extends Component {
	render() {
		const {items} = this.props;

		return (
			<div className={'container-fluid'}>
				<div className={'row no-gutter'}>
					{
						items.map((item) => {
							const {_id, name, price, rarity, type, imageSrc} = item;

							return (
								<div
									key={_id}
									className={'col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3'}
								>
									<ItemCard
										_id={_id}
										name={name}
										price={price}
										rarity={rarity}
										type={type}
										imageSrc={imageSrc}
									/>
								</div>
							);
						})
					}
				</div>
			</div>
		);
	}
}

ItemsList.propTypes = {
	items: PropTypes.array
};

export default ItemsList;