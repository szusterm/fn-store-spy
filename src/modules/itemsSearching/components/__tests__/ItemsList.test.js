import React from 'react';
import {shallow} from 'enzyme';

import {ItemsList} from '../ItemsList/ItemsList';

import ItemCard from '../ItemSearchCard';

const exampleItemsList = [
	{
		_id: '3h5i34uh53i4',
		fnbrId: '6u69uj6kuj',
		name: 'Super Axe',
		price: '2000',
		rarity: 'uncommon',
		type: 'axe',
		imageSrc: 'https://example.com'
	},
	{
		_id: '56b7h8678b',
		fnbrId: '4395ytj4f905',
		name: 'Something',
		price: '800',
		rarity: 'rare',
		type: 'boots',
		imageSrc: 'https://example.pl'
	}
];

const setup = (propOverrides) => {
	const props = Object.assign({
		items: exampleItemsList
	}, propOverrides);

	const wrapper = shallow(<ItemsList {...props}/>);

	return {props, wrapper};
};

describe('ItemsList Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('shows items of ItemCards from prop', () => {
		const {wrapper, props} = setup({
			items: exampleItemsList
		});

		const items = wrapper.find(ItemCard);

		expect(items.length).toBe(props.items.length);
	});

	it('passes _id as a key prop to every ItemSearchCard', () => {
		const {wrapper, props} = setup({
			items: exampleItemsList
		});

		for (const [index, itemProps] of props.items.entries()) {
			const itemCardIns = wrapper.find(ItemCard).at(index);
			expect(itemCardIns.key()).toEqual(itemProps._id);
		}
	});

	it('passes props to every ItemSearchCard', () => {
		const {wrapper, props} = setup({
			items: exampleItemsList
		});

		for (const [index, itemProps] of props.items.entries()) {
			delete itemProps._id;

			const itemCardIns = wrapper.find(ItemCard).at(index);
			expect(itemCardIns.props()).toEqual({...itemProps});
		}
	});
});