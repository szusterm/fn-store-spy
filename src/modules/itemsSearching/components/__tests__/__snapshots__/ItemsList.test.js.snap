// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemsList
    items={
      Array [
        Object {
          "_id": "3h5i34uh53i4",
          "fnbrId": "6u69uj6kuj",
          "imageSrc": "https://example.com",
          "name": "Super Axe",
          "price": "2000",
          "rarity": "uncommon",
          "type": "axe",
        },
        Object {
          "_id": "56b7h8678b",
          "fnbrId": "4395ytj4f905",
          "imageSrc": "https://example.pl",
          "name": "Something",
          "price": "800",
          "rarity": "rare",
          "type": "boots",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container-fluid"
      >
        <div
          className="row no-gutter"
        >
          <Connect(ItemSearchCard)
            fnbrId="6u69uj6kuj"
            imageSrc="https://example.com"
            name="Super Axe"
            price="2000"
            rarity="uncommon"
            type="axe"
          />
          <Connect(ItemSearchCard)
            fnbrId="4395ytj4f905"
            imageSrc="https://example.pl"
            name="Something"
            price="800"
            rarity="rare"
            type="boots"
          />
        </div>
      </div>,
      "className": "col-12 col-sm-12 col-md-11 col-lg-10 col-xl-8",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row no-gutter"
        >
          <Connect(ItemSearchCard)
            fnbrId="6u69uj6kuj"
            imageSrc="https://example.com"
            name="Super Axe"
            price="2000"
            rarity="uncommon"
            type="axe"
          />
          <Connect(ItemSearchCard)
            fnbrId="4395ytj4f905"
            imageSrc="https://example.pl"
            name="Something"
            price="800"
            rarity="rare"
            type="boots"
          />
        </div>,
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(ItemSearchCard)
              fnbrId="6u69uj6kuj"
              imageSrc="https://example.com"
              name="Super Axe"
              price="2000"
              rarity="uncommon"
              type="axe"
            />,
            <Connect(ItemSearchCard)
              fnbrId="4395ytj4f905"
              imageSrc="https://example.pl"
              name="Something"
              price="800"
              rarity="rare"
              type="boots"
            />,
          ],
          "className": "row no-gutter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "3h5i34uh53i4",
            "nodeType": "class",
            "props": Object {
              "fnbrId": "6u69uj6kuj",
              "imageSrc": "https://example.com",
              "name": "Super Axe",
              "price": "2000",
              "rarity": "uncommon",
              "type": "axe",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "56b7h8678b",
            "nodeType": "class",
            "props": Object {
              "fnbrId": "4395ytj4f905",
              "imageSrc": "https://example.pl",
              "name": "Something",
              "price": "800",
              "rarity": "rare",
              "type": "boots",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container-fluid"
        >
          <div
            className="row no-gutter"
          >
            <Connect(ItemSearchCard)
              fnbrId="6u69uj6kuj"
              imageSrc="https://example.com"
              name="Super Axe"
              price="2000"
              rarity="uncommon"
              type="axe"
            />
            <Connect(ItemSearchCard)
              fnbrId="4395ytj4f905"
              imageSrc="https://example.pl"
              name="Something"
              price="800"
              rarity="rare"
              type="boots"
            />
          </div>
        </div>,
        "className": "col-12 col-sm-12 col-md-11 col-lg-10 col-xl-8",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row no-gutter"
          >
            <Connect(ItemSearchCard)
              fnbrId="6u69uj6kuj"
              imageSrc="https://example.com"
              name="Super Axe"
              price="2000"
              rarity="uncommon"
              type="axe"
            />
            <Connect(ItemSearchCard)
              fnbrId="4395ytj4f905"
              imageSrc="https://example.pl"
              name="Something"
              price="800"
              rarity="rare"
              type="boots"
            />
          </div>,
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(ItemSearchCard)
                fnbrId="6u69uj6kuj"
                imageSrc="https://example.com"
                name="Super Axe"
                price="2000"
                rarity="uncommon"
                type="axe"
              />,
              <Connect(ItemSearchCard)
                fnbrId="4395ytj4f905"
                imageSrc="https://example.pl"
                name="Something"
                price="800"
                rarity="rare"
                type="boots"
              />,
            ],
            "className": "row no-gutter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "3h5i34uh53i4",
              "nodeType": "class",
              "props": Object {
                "fnbrId": "6u69uj6kuj",
                "imageSrc": "https://example.com",
                "name": "Super Axe",
                "price": "2000",
                "rarity": "uncommon",
                "type": "axe",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "56b7h8678b",
              "nodeType": "class",
              "props": Object {
                "fnbrId": "4395ytj4f905",
                "imageSrc": "https://example.pl",
                "name": "Something",
                "price": "800",
                "rarity": "rare",
                "type": "boots",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
